sstables/index_reader.hh

```diff
@@ -527,6 +528,15 @@ private:
                     if (ex) {
                         return make_exception_future<index_list>(std::move(ex));
                     }
+                    if (_single_page_read) {
+                        // if the associated reader is forwarding despite having singular range, we prepare for that
+                        _single_page_read = false;
+                        return do_with(std::move(bound.context), [this] (std::unique_ptr<index_consume_entry_context<index_consumer>>& moved_context) {
```

Comment:
We could avoid the allocation incurred by do_with():

```
auto& ctx = *bound.context;
return ctx.close().then([bc = std::move(bound.context)] {});
```

Reply:
Ok